calyx/src/frontend/syntax.pest

```diff
@@ -231,7 +235,10 @@ enable = { at_attributes ~ identifier ~ ";" }
 
 invoke_arg = { identifier ~ "=" ~ (port | num_lit) }
 invoke_args = { (invoke_arg ~ ("," ~ invoke_arg)*)? }
-invoke = { at_attributes ~ "invoke" ~ identifier ~ "(" ~ invoke_args ~ ")" ~ "(" ~ invoke_args ~ ")" ~ ("with" ~ identifier)? ~ ";" }
+invoke_external_arg = {identifier ~ "=" ~ identifier}
+invoke_external_args = {(invoke_external_arg ~ ("," ~ invoke_external_arg)*)?}
+ivk_extern_args_in_paren = {(sq_paren_l ~ invoke_external_args ~ sq_paren_r)?}
```

Comment:
This encoding can be simplified a bit:
1. You can prefix `_` before `invoke_external_arg` to make it a [silent rule](https://pest.rs/book/grammars/syntax.html#silent-and-atomic-rules). A silent rule is simply syntactic sugar and does not generate a parser in the `parser.rs` file.
2. You can also do the same to `invoke_external_args` so that the only parsing rule is `ivk_extern_args_in_paren`.
3. Why does this need to use the `sq_paren_l` and `sq_paren_r` parsers instead of just `"["` and `"]"`?